app/controllers/api/v1/password.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module API
+  module V1
+    class Password < Grape::API
+      helpers Doorkeeper::Grape::Helpers
+
+      before do
+        doorkeeper_authorize!
+
+        def current_account
+          Account.find(doorkeeper_token.resource_owner_id) if doorkeeper_token
+        end
+      end
+
+      desc 'Password related routes'
+      resource :password do
+        desc 'Change user\'s passord'
+        post '/update' do
+          account = current_account
```

Comment:
Use just `put do ... end`
Use just current_account
Declare params. Don't use old_password. Doorkeeper has already verified it
Use allow_blank: false in params

Your save! will raise error and you don't handle it.
```
unless current_account.update(declared(params)
  error!(account.errors.full_messages.to_sentence)
end
```